src/online2bin/ivector-extract-online2.cc

```diff
@@ -98,43 +117,49 @@ int main(int argc, char *argv[]) {
           continue;
         }
         const Matrix<BaseFloat> &feats = feature_reader.Value(utt);
-        
-        OnlineMatrixFeature matrix_feature(feats);
-
-        OnlineIvectorFeature ivector_feature(ivector_info,
-                                             &matrix_feature);
-        
-        ivector_feature.SetAdaptationState(adaptation_state);
 
         int32 T = feats.NumRows(),
             n = (repeat ? 1 : ivector_config.ivector_period),
             num_ivectors = (T + n - 1) / n;
-        
-        Matrix<BaseFloat> ivectors(num_ivectors,
-                                   ivector_feature.Dim());
-        
-        for (int32 i = 0; i < num_ivectors; i++) {
-          int32 t = i * n;
-          SubVector<BaseFloat> ivector(ivectors, i);
-          ivector_feature.GetFrame(t, &ivector);
+
+         Matrix<BaseFloat> ivectors(num_ivectors,
+                                   num_cmn_offset * ivec_dim);
+
+        for (size_t offset_ind = 0; offset_ind < num_cmn_offset; offset_ind++) { 
+          Matrix<BaseFloat> shifted_feats(feats);
+          if (cmn_offset != NULL)
```

Comment:
Please be more careful about avoiding unnecessary copies of variables.
You could rename 'feats' to 'feats_orig', then do:
```
  Matrix<BaseFloat> shifted_feats;
  const Matrix<BaseFloat> &feats = (cmn_offset != NULL ? shifted_feats : feats_orig);
  if (cmn_offset != NULL) {
    shifted_feats = feats;
    shifted_feats.AddVecToRows(1.0, cmn_offset->Row(offset));
  }
```
... and make sure line length <= 80.